source/brailleDisplayDrivers/albatross/driver.py

```diff
@@ -205,6 +213,24 @@ def _searchPorts(self, port: str):
 		else:
 			raise RuntimeError("No Albatross found")
 
+	def _albatross_port(self, port: str) -> bool:
+		"""Check if this is albatross usb serial port or other suitable serial port
+
+		:param port: port to check
+		:return: False, if correct vid and pid but bus device description and
+		serial number do not match; otherwise True
+		"""
+		p = next(list_ports.grep(port))
+		if hex(p.vid) == ALBATROSS_VID and hex(p.pid) == ALBATROSS_PID:
+			for entry in ftdi2.get_device_info_list():

```

Comment:
Note that this fetches all the ftdi ports again for every encountered port. Could you avoid this?